specification/sigma-appendix-custom-taxonomy.md

```diff
@@ -0,0 +1,96 @@
+
+```yml
+category: network
+service: connection
+```
+
+The field names follow the field names used in [Elastic Common Schema](https://www.elastic.co/guide/en/ecs/current/ecs-network.html) events `taxonomy: ecs`:
+
+| Field Name           | Example Value                  | Comment                                           |
+| -------------------- | ------------------------------ | ------------------------------------------------- |
+| @timestamp           | 2025-05-08T14:38:35.000Z       | Event timestamp in UTC                            |
+| event.duration       | 0.005726                       | Connection duration in seconds                    |
+| event.id             | CXWfMc4eWKNBm1O4fl             | Unique connection identifier                      |
+| network.type         | ipv4                           | Network layer type (e.g., ipv4, ipv6, ipsec)      |
+| network.transport    | tcp                            | Transport layer protocol (e.g., tcp, udp)         |
+| network.protocol     | http                           | Application layer protocol (e.g., http, dns, ssh) |
+| source.ip            | 192.168.1.100                  | Source IP address                                 |
+| source.port          | 54321                          | Source port number                                |
+| destination.ip       | 93.184.216.34                  | Destination IP address                            |
+| destination.port     | 443                            | Destination port number                           |
+| source.packets       | 98                             | Number of packets                                 |
+| destination.packets  | 45                             | Number of packets                                 |
+| source.bytes         | 2159                           | Number of bytes                                   |
+| destination.bytes    | 4739                           | Number of bytes                                   |
+| network.community_id | 1:LQU9qZlK+B5F3KDmev6m5PMibrg= | Community ID hash                                 |
+| network.state        | SF                             | State of the connection                           |
+| network.history      | shADd                          | History of the connection                         |
+
+### Network DNS
+
+This logsource covers DNS queries in general.
+
+```yml
+category: network
+service: dns
+```
+
+The field names follow the field names used in [Elastic Common Schema](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html) events `taxonomy: ecs`:
+
+| Field Name           | Example Value                  | Comment                      |
+| -------------------- | ------------------------------ | ---------------------------- |
+| @timestamp           | 2025-05-08T14:38:35.000Z       | Event timestamp in UTC       |
+| event.id             | CXWfMc4eWKNBm1O4fl             | Unique connection identifier |
```

Comment:
See above, usually absolute timestamps and unique identifiers are not very useful in detection rules.